### PR TITLE
Add warning on lesson plan if parts of lesson require being verified and teacher is not verified

### PR DESCRIPTION
**Pull request by dmcavoy**
Show warning on lesson plan if the lesson has verified resources or is lockable and the teacher is not verified.

Screen Shot 2021-04-13 at 11 57 42 PM

## Links

Part 1 of: https://codedotorg.atlassian.net/browse/PLAT-968

## Testing story

- New unit tests for the different components
- Manually tested

### CODE REVIEW

File: apps/src/templates/courseOverview/VerifiedResourcesNotification.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-      type={NotificationType.information}
-      notice={i18n.verifiedResourcesNotice()}
-      details={i18n.verifiedResourcesDetails()}
-      buttonText={i18n.learnMore()}
-      buttonLink="https://support.code.org/hc/en-us/articles/115001550131"
-      dismissible={true}
-      width={width}
-    />
-  );
+export default class VerifiedResourcesNotification extends PureComponent {
```
Comment (bethanyaconnor) [inline apps/src/templates/courseOverview/VerifiedResourcesNotification.js:6]: For my own knowledge: why did you change this from a function to a PureComponent?

File: apps/src/templates/lessonOverview/LessonOverview.jsx [hunk 2]
```diff
@@ -73,11 +74,24 @@ class LessonOverview extends Component {
...
+    const {
+      lesson,
+      announcements,
+      isSignedIn,
+      viewAs,
+      isVerifiedTeacher,
+      hasVerifiedResources
+    } = this.props;
+
+    const displayVerifiedResources =
```
Comment (bethanyaconnor) [inline apps/src/templates/lessonOverview/LessonOverview.jsx]: nit: `displayVerifiedResources` sounds like this is a flag to display the resources not the notification. Could it be `displayVerifiedResourcesNotification` or something instead?

File: dashboard/app/models/lesson.rb [hunk 3]
```diff
@@ -707,6 +709,11 @@ def resources_for_lesson_plan(verified_teacher)
     grouped_resources
   end
 
+  def lesson_plan_has_verified_resources
+    grouped_resources = resources.map(&:summarize_for_lesson_plan).group_by {|r| r[:audience]}
```
Comment (bethanyaconnor) [inline dashboard/app/models/lesson.rb]: nit: if you don't need to group the resources could you do something like `resources.any? {|r| r.audience == 'Verified Teacher'}`?

**Review by bethanyaconnor — APPROVED** [review]
A couple nits and question for my own understanding but looks good!

File: apps/src/templates/lessonOverview/LessonOverview.jsx (hunk 2 above)
Comment (dmcavoy) [reply apps/src/templates/lessonOverview/LessonOverview.jsx]: Done!

File: dashboard/app/models/lesson.rb (hunk 3 above)
Comment (dmcavoy) [reply dashboard/app/models/lesson.rb]: Done!

File: apps/src/templates/courseOverview/VerifiedResourcesNotification.js (hunk 1 above)
Comment (dmcavoy) [reply apps/src/templates/courseOverview/VerifiedResourcesNotification.js:6]: It just seemed more in line with how it was being used especially when I added tests for it.
